### ISSUE TITLE
For loop error

### ISSUE THREAD

**Issue by anhh** [NeilFraser/JS-Interpreter#126]
Below code suppose to show 2 alert messages, but there are 3 alert messages.
Last alert message is constructor.

var x = {a:1, b:2};
for (var i in x)
{
  alert(i);
}

**Comment by anhh** [comment]
interpreter.js 
if "Interpreter.prototype['stepForInStatement']" function 3008 line "prop" variable has no value,
then exit of the for loop?

sorry for my bad english.

**Comment by cpcallen** [comment]
This is interesting: the third alerted value is 'constructor'.

I tried `alert(Object.getOwnPropertyDescriptor(Object.prototype, 'constructor').enumerable);`, which resulted in:

```
Uncaught RangeError: Maximum call stack size exceeded
    at Interpreter.nativeToPseudo (interpreter.js:1782)
    at Interpreter.nativeToPseudo (interpreter.js:1826)
    at Interpreter.nativeToPseudo (interpreter.js:1826)
    at Interpreter.nativeToPseudo (interpreter.js:1826)
```
(etc…)

More investigation required.

**Comment by islemaster** [comment]
For what it's worth, this error doesn't repro in our fork at https://github.com/code-dot-org/JS-Interpreter/commit/94148ca744050a445e561d4507b101d89c8d745c so this is probably a regression introduced somewhere [in this subset of 39 commits](https://github.com/code-dot-org/JS-Interpreter/compare/master...NeilFraser:master).

**Comment by NeilFraser** [comment]
Pretty sure this is a real bug.  Reopening.

**Comment by NeilFraser** [comment]
Fixed in 1b6821914bac00786b76b37b69115ba0e054ec82.  Thanks!
